docs/integrations/sources/microsoft-sharepoint.md

```diff
@@ -0,0 +1,134 @@
+# Microsoft SharePoint
+<HideInUI>
+
+This page contains the setup guide and reference information for the Microsoft SharePoint source connector.
+
+</HideInUI>
+### Requirements
+
+* Application \(client\) ID 
+* Directory \(tenant\) ID
+* Drive name
+* Folder Path
+* Client secrets 
+
+## Setup guide
+
+<!-- env:cloud -->
+<HideInUI>
+
+**For Airbyte Cloud:**
+
+</HideInUI>
+1. Navigate to the Airbyte Open Source dashboard.
+2. Click **Sources** and then click **+ New source**.
+3. On the Set up the source page, select **Microsoft SharePoint** from the Source type dropdown.
+4. Enter the name for the Microsoft SharePoint connector.
+5. Enter **Drive Name**. To find your drive name go to settings and at the top of setting menu you can find the name of your drive.
+6. Enter **Folder Path**. 
+7. The **OAuth2.0** authorization method is selected by default. Click **Authenticate your Microsoft SharePoint account**. Log in and authorize your Microsoft account.
+8. For **Start Date**, enter the date in YYYY-MM-DD format. The data added on and after this date will be replicated.
+9. Add a stream:
+   1. Write the **File Type**
+   2. In the **Format** box, use the dropdown menu to select the format of the files you'd like to replicate. The supported formats are **CSV**, **Parquet**, **Avro** and **JSONL**. Toggling the **Optional fields** button within the **Format** box will allow you to enter additional configurations based on the selected format.  For a detailed breakdown of these settings, refer to the [File Format section](#file-format-settings) below.
+   3. Give a **Name** to the stream
+   4. (Optional) - If you want to enforce a specific schema, you can enter a **Input schema**. By default, this value is set to `{}` and will automatically infer the schema from the file\(s\) you are replicating. For details on providing a custom schema, refer to the [User Schema section](#user-schema).
+   5. Optionally, enter the **Globs** which dictates which files to be synced. This is a regular expression that allows Airbyte to pattern match the specific files to replicate. If you are replicating all the files within your bucket, use `**` as the pattern. For more precise pattern matching options, refer to the [Path Patterns section](#path-patterns) below.
+10. Click **Set up source**
+<!-- /env:cloud -->
+
+<!-- env:oss -->
+<HideInUI>
+
+**For Airbyte Open Source:**
+
+</HideInUI>
+### Step 1: Set up SharePoint application
+
+The Microsoft Graph API uses OAuth for authentication. Microsoft Graph exposes granular permissions that control the access that apps have to resources, like users, groups, and mail. When a user signs in to your app they, or, in some cases, an administrator, are given a chance to consent to these permissions. If the user consents, your app is given access to the resources and APIs that it has requested. For apps that don't take a signed-in user, permissions can be pre-consented to by an administrator when the app is installed.
```

Comment:
nit: "When a user signs in to your app, they or in some cases an administrator are given". This would help with the flow of the sentence

Reply:
Fixed